ballista/core/src/execution_plans/coalesce_tasks.rs

```diff
@@ -112,6 +112,11 @@ impl ExecutionPlan for CoalesceTasksExec {
         let elapsed_compute = baseline_metrics.elapsed_compute().clone();
         let _timer = elapsed_compute.timer();
 
+        // If only executing a single partition, execute it directly
```

Comment:
Is this needed? This could also be in the planner/optimizer (don't convert to `CoalesceTasks`)?